tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,214 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, LabeledList, Section, ProgressBar, Flex, Box, Table, Collapsible, Input, Dimmer, Icon } from '../components';
+import { Window } from '../layouts';
+import { capitalize } from "common/string";
+
+export const Autolathe = (props, context) => {
+  const { act, data } = useBackend(context);
+  // Extract `health` and `color` variables from the `data` object.
+  const {
+    materialtotal,
+    materialsmax,
+    materials = [],
+    categories = [],
+    designs = [],
+    hasDisk,
+    active,
+  } = data;
+  const [
+    current_category,
+    setCategory,
+  ] = useLocalState(context, 'current_category', "None");
+  const filteredmaterials = materials.filter(material =>
+    material.mineral_amount > 0);
+  return (
+    <Window
+      title="Autolathe"
+      width={600}
+      height={600}>
+      <Window.Content scrollable>
+        <Section
+          title="Total Materials"
+          buttons={(
+            <Button
+              icon="eject"
+              content="Eject design disk"
+              disabled={!hasDisk}
+              onClick={() => {
+                  act('diskEject');
```

Comment:
```suggestion
                act('diskEject');
```
This'll fix that error